cmd/encryption-v1.go

```diff
@@ -141,55 +149,70 @@ const (
 	ServerSideEncryptionSealedKey = ReservedMetadataPrefix + "Server-Side-Encryption-Sealed-Key"
 )
 
-// SSESealAlgorithmDareSha256 specifies DARE as authenticated en/decryption scheme and SHA256 as cryptographic
-// hash function.
-const SSESealAlgorithmDareSha256 = "DARE-SHA256"
+const (
+	// SSEIVSize is the size of the randomly chosen IV used to derive the key-encryption-key.
+	SSEIVSize = 32
+
+	// DAREPayloadSize is the max. size of the package payload for a DARE 2.0 package.
+	DAREPayloadSize = 1 << 16
+
+	// DAREOverhead is the overhead (header + tag) added by encrypting one payload using DARE 2.0.
+	DAREOverhead = 32
 
-// hasSSECustomerHeader returns true if the given HTTP header
+	// SSESealAlgorithmDareSha256 specifies DARE as authenticated en/decryption scheme and SHA256 as
+	// cryptographic hash function.
+	SSESealAlgorithmDareSha256 = "DARE-SHA256"
+)
+
+// ContainsSSECustomerHeader returns true if the given HTTP header
 // contains server-side-encryption with customer provided key fields.
-func hasSSECustomerHeader(header http.Header) bool {
+func ContainsSSECustomerHeader(header http.Header) bool {
 	return header.Get(SSECustomerAlgorithm) != "" || header.Get(SSECustomerKey) != "" || header.Get(SSECustomerKeyMD5) != ""
 }
 
-// hasSSECopyCustomerHeader returns true if the given HTTP header
+// ContainsSSECopyCustomerHeader returns true if the given HTTP header
 // contains copy source server-side-encryption with customer provided key fields.
-func hasSSECopyCustomerHeader(header http.Header) bool {
+func ContainsSSECopyCustomerHeader(header http.Header) bool {
 	return header.Get(SSECopyCustomerAlgorithm) != "" || header.Get(SSECopyCustomerKey) != "" || header.Get(SSECopyCustomerKeyMD5) != ""
 }
 
-// ParseSSECopyCustomerRequest parses the SSE-C header fields of the provided request.
-// It returns the client provided key on success.
-func ParseSSECopyCustomerRequest(r *http.Request) (key []byte, err error) {
+// parseSSECustomerHeaders parses the SSE-C header fields of the provided headers depending
+// on the three provided S3 sse headers (either regular request or copy request).
+// It erases the client key form the HTTP headers.
+//
+// It also returns an error if (valid) SSE-C headers are provided but the server does not run
+// in TLS mode.
+func parseSSECustomerHeaders(header http.Header, sseHeaders [3]string) (key []byte, err error) {
```

Comment:
In my opinion, it is better to create a new struct, we can call it sseMaterials or something else, and we pass it to this function instead of array of string. That struct will have algo, key, keyMD5 members. What do you think @aead ?

Reply:
s/sseMaterials/sseHeaders/
s/algo/algoHdr
s/key/keyHdr
s/keyMD5/keyMD5Hdr

Reply:
We can - a dedicated struct would be more explicit (you "cannot" mix eg. key with keyMD5 by accident) - it is not really useful except for grouping values but I think here it's fine.